src/analytics/private/scan-metadata-workflow.ts

```diff
@@ -0,0 +1,274 @@
+  }
+
+  private createWorkflow(props: ScanMetadataWorkflowProps): IStateMachine {
+    const getJobList = new Pass(this, `${this.node.id} - Get app_id`, {
+      parameters: {
+        Payload: {
+          'appIdList.$': `States.StringSplit('${props.appIds}', ',')`,
+        },
+      },
+      outputPath: '$.Payload',
+    });
+
+    const scanMetadataFn = this.scanMetadataFn(props);
+    const scanMetadataJob = new LambdaInvoke(this, `${this.node.id} - Submit scan metadata job`, {
+      lambdaFunction: scanMetadataFn,
+      payload: TaskInput.fromObject({
+        detail: {
+          'appId.$': '$',
+        },
+      }),
+      outputPath: '$.Payload',
+    });
+
+    const checkScanMetadataStatusFn = this.checkScanMetadataStatusFn(props);
+    const checkScanMetadataStatusJob = new LambdaInvoke(this, `${this.node.id} - Check scan metadata job status`, {
+      lambdaFunction: checkScanMetadataStatusFn,
+      payload: TaskInput.fromObject({
+        'detail.$': '$.detail',
+      }),
+      outputPath: '$.Payload',
+    });
+
+    const waitX = new Wait(this, `${this.node.id} - Wait seconds`, {
+      /**
+         *  You can also implement with the path stored in the state like:
+         *  sfn.WaitTime.secondsPath('$.waitSeconds')
+      */
+      time: WaitTime.duration(Duration.seconds(30)),
+    });
+
+    const scanMetadataJobFailed = new Fail(this, `${this.node.id} - scan metadata job fails`, {
+      cause: 'ScanMetadata Job Failed',
+      error: 'DescribeJob returned FAILED',
+    });
+
+    const scanMetadataCompleted = new Succeed(this, `${this.node.id} - scan metadata job completes`);
+
+    const storeMetadataIntoDDBFn = this.storeMetadataIntoDDBFn(props);
+    const storeMetadataIntoDDBJob = new LambdaInvoke(this, `${this.node.id} - store scan metadata job status`, {
+      lambdaFunction: storeMetadataIntoDDBFn,
+      payload: TaskInput.fromObject({
+        'detail.$': '$.detail',
+      }),
+      outputPath: '$.Payload',
+    }).next(
+      new Choice(this, 'Check if store metadata completed')
+        .when(Condition.stringEquals('$.detail.status', 'SUCCEED'), scanMetadataCompleted)
+        .otherwise(scanMetadataJobFailed),
+    );
+
+    // Create sub chain
+    const subDefinition = scanMetadataJob
+      .next(waitX)
+      .next(checkScanMetadataStatusJob)
+      .next(
+        new Choice(this, `${this.node.id} - Check if scan metadata job completes`)
+          .when(Condition.stringEquals('$.detail.status', 'FAILED'), scanMetadataJobFailed)
+          .when(Condition.stringEquals('$.detail.status', 'ABORTED'), scanMetadataJobFailed)
+          .when(Condition.stringEquals('$.detail.status', 'FINISHED'), storeMetadataIntoDDBJob)
+          .when(Condition.stringEquals('$.detail.status', 'NO_JOBS'), scanMetadataCompleted)
+          .otherwise(waitX));
+
+    const doScanMetadataJob = new Map(
+      this,
+      `${this.node.id} - Do scanMetadata job`,
+      {
+        maxConcurrency: 1,
```

Comment:
why not allowing concurrency to scan the metadata for different apps?

Reply:
be consistent with upsert user workflow.
For multi appIds, the workflow will be executed sequentially one by one.